packages/rich-text/src/plugins/Quote/index.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import * as Slate from 'slate-react';
+import { css } from 'emotion';
+import tokens from '@contentful/forma-36-tokens';
+import { EditorToolbarButton } from '@contentful/forma-36-react-components';
+import { Transforms, Editor, Node, Path } from 'slate';
+import { BLOCKS } from '@contentful/rich-text-types';
+import { useCustomEditor } from '../../hooks/useCustomEditor';
+import { CustomElement, CustomEditor } from 'types';
+
+const styles = {
+  blockquote: css({
+    margin: '0 0 1.3125rem',
+    borderLeft: `6px solid ${tokens.colorElementLight}`,
+    paddingLeft: '0.875rem',
+    fontStyle: 'normal',
+    '& a': {
+      color: 'inherit',
+    },
+  }),
+};
+
+interface ToolbarQuoteButtonProps {
+  isDisabled?: boolean;
+}
+
+const createBlockQuote = (editor: CustomEditor) => {
+  if (!editor.selection) return;
+
+  const text = { text: '' };
+  const paragraph = { type: BLOCKS.PARAGRAPH, children: [text] };
+  const path = editor.selection.focus.path;
+  const parent = Editor.parent(editor, path);
+  const next = Editor.next(editor, { at: parent[1] });
+
+  editor.toggleBlock(BLOCKS.QUOTE);
+
+  if (!next) {
+    const next = Path.next(parent[1]);
+    Transforms.insertNodes(editor, paragraph, { at: next });
+  }
+};
+
+export function withQuoteEvents(editor: CustomEditor, event: KeyboardEvent) {
+  if (!editor.selection) return;
+
+  const [currentFragment] = Editor.fragment(editor, editor.selection.focus.path) as CustomElement[];
+  const isEnter = event.keyCode === 13;
+
+  if (isEnter && currentFragment?.type === BLOCKS.QUOTE) {
+    event.preventDefault();
+
+    const text = { text: '' };
+    const paragraph = { type: BLOCKS.PARAGRAPH, children: [text] };
+
+    if (editor.hasSelectionText()) {
+      const currentOffset = editor.selection.focus.offset;
+      const currentTextLength = Node.string(currentFragment).length;
+      const cursorIsAtTheBeginning = currentOffset === 0;
+      const cursorIsAtTheEnd = currentTextLength === currentOffset;
```

Comment:
Maybe we should add this logic into the `Editor`? We are using the same thing on the headings and I will probably use the same thing for the lists.